### PR TITLE
fix: Ignore NRF URI containing a specific IP on registration

### LINKED ISSUE #51: UDM does not handle NRF IP changes
The UDM does not recover on its own if the NRF it is registered to changes IP for any reason, including a pod crashing. To reproduce the issue, simply delete the NRF pod, and once it is recreated, the UDM will continually log lines similar to:

```
2023-11-22T20:30:58.957Z [udm] UDM register to NRF Error[Put "https://10.1.33.88:29510/nnrf-nfm/v1/nf-instances/d4936127-edb3-4b87-98f3-e82c48845dd5": dial tcp 10.1.33.88:29510: connect: connection timed out]
```

The IP in the log line will be the IP of the previous pod.

A workaround is to also delete the UDM pod. The initialization will fix the issue until the NRF changes IP again.

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
This ignores the NRF URI received from the NRF on registration, as the NRF will return its IP, which is subject to change. This is similar to what is done on the AMF.

Fixes #51

### CODE REVIEW

**Comment by ghislainbourgeois** [comment]
Setting to draft as this is missing tests.

File: service/init.go [hunk 1]
```diff
@@ -435,16 +435,14 @@ func (udm *UDM) registerNF() {
 		if err != nil {
 			logger.InitLog.Errorln(err.Error())
 		} else {
-			var newNrfUri string
 			var err1 error
 			var prof models.NfProfile
-			prof, newNrfUri, self.NfId, err1 = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, proflie)
+			prof, _, self.NfId, err1 = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, proflie)
```
Comment (gatici) [inline service/init.go:440]: we can also correct typo proflie -> profile

Comment (ghislainbourgeois) [reply service/init.go]: Done, nice catch!

**Comment by gab-arrobo** [comment]
test this please

**Comment by gab-arrobo** [comment]
test this please

**Review by gab-arrobo — APPROVED** [review]
+1 (Verified changes with make test and AiaB)

**Pull request merged**